sql/core/src/main/java/org/apache/spark/sql/sources/v2/vectorized/ColumnarBatch.java

```diff
@@ -87,19 +79,7 @@ public void remove() {
   }
 
   /**
-   * Resets the batch for writing.
-   */
-  public void reset() {
-    for (int i = 0; i < numCols(); ++i) {
-      if (columns[i] instanceof WritableColumnVector) {
-        ((WritableColumnVector) columns[i]).reset();
-      }
-    }
-    this.numRows = 0;
```

Comment:
This can result an incorrect `numRows`. May it be a potential error?

Reply:
This doesn't matter. The `numRows` is only used when calling `rowsInterator`, and we always call `setNumRows` before calling `rowsIterator`